src/mongocxx/database.cpp

```diff
@@ -42,6 +42,13 @@ database::database(const class client& client, stdx::string_view name)
           name.data())) {
 }
 
+database::database(const database& d) : _impl{stdx::make_unique<impl>(*(d._impl))} { }
+
+database& database::operator=(const database& d) {
+    _impl = stdx::make_unique<impl>(*(d._impl));
```

Comment:
if we wanted to use the copy assignment operator of the impl this line would be

*impl = *d._impl. I think this is better though so our old impl gets cleaned up automatically